### PR TITLE
[0.12] Rename permitrbf to mempoolreplacement and provide minimal string-list forward compatibility

### PR DESCRIPTION
**Pull request by luke-jr**
Backport of #7431

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -144,13 +144,13 @@ accepted when it pays sufficient fee, as described in [BIP 125]
 (https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki).
 
 Transaction replacement can be disabled with a new command line option,
-`-permitrbf=false`.  Transactions signaling replacement under BIP125 will still
```
Comment (luke-jr) [inline doc/release-notes.md]: @sdaftuar Note that "false" and "true" are invalid for boolean options!

**Comment by MarcoFalke** [comment]
@luke-jr What about doing the same like 89d113e02a83617b4e971c160d47551476dacc71?

**Comment by sandakersmann** [comment]
I think replacebyfee gives a better description of what this is.

**Comment by dcousens** [comment]
@sandakersmann except the idea is that you may be able to change the policy to things other than RBF.  Such as some special cost function that balances priority and fees and SOMEFACTOR(TM).

**Comment by dcousens** [comment]
Nevermind,

**Comment by luke-jr** [comment]
@sandakersmann The idea is that future or derivative versions may support more replacement logic than merely by fee. The original replacement feature was planned to simply go based on sequence number, but that had DoS risks - RBF solves that, but it's not impossible someone may come up with an alternative solution. Anyway, please use #7431 for comments on the change itself - this PR is merely a backport.

**Pull request merged**